format/jsontoinflux10.go

```diff
@@ -0,0 +1,175 @@
+// JsonToInfluxFormatter defines the formatter applied before the conversion
+// from JSON to InfluxDB. By default this is set to format.Forward.
+type JSONToInflux10 struct {
+	core.SimpleFormatter `gollumdoc:"embed_type"`
+	metadataEscape       *strings.Replacer
+	measurementEscape    *strings.Replacer
+	fieldValueEscape     *strings.Replacer
+	timeField            string `config:"Timefield" default:"time"`
+	measurement          string `config:"Measurement" default:"measurement"`
+	tags                 map[string]bool
+	ignore               map[string]bool
+}
+
+func init() {
+	core.TypeRegistry.Register(JSONToInflux10{})
+}
+
+// Configure initializes this formatter with values from a plugin config.
+func (format *JSONToInflux10) Configure(conf core.PluginConfigReader) {
+	format.SimpleFormatter.Configure(conf)
+
+	format.metadataEscape = strings.NewReplacer(",", "\\,", "=", "\\=", " ", "\\ ")
+	format.measurementEscape = strings.NewReplacer(",", "\\,", " ", "\\ ")
+	format.fieldValueEscape = strings.NewReplacer("\"", "\\\"")
+
+	// Create a set from the tags array for O(1) lookup
+	tagsArray := conf.GetStringArray("Tags", []string{})
+	format.tags = map[string]bool{}
+	for _, v := range tagsArray {
+		format.tags[v] = true
+	}
+
+	// Create a set from the ignore array for O(1) lookup
+	ignoreArray := conf.GetStringArray("Ignore", []string{})
+	format.ignore = map[string]bool{}
+	for _, v := range ignoreArray {
+		format.ignore[v] = true
+	}
+}
+
+func (format *JSONToInflux10) escapeMeasurment(value string) string {
+	return format.measurementEscape.Replace(value)
+}
+
+func (format *JSONToInflux10) escapeMetadata(value string) string {
+	return format.metadataEscape.Replace(value)
+}
+
+func (format *JSONToInflux10) escapeFieldValue(value string) string {
+	return format.fieldValueEscape.Replace(value)
+}
+
+func (format *JSONToInflux10) joinMap(m map[string]interface{}) string {
+	var tokens []string
+	for k, v := range m {
+		tokens = append(tokens, fmt.Sprintf(`%s=%s`, k, v))
+	}
+	return fmt.Sprintf(strings.Join(tokens, ","))
+}
+
+// ApplyFormatter updates the message payload
+func (format *JSONToInflux10) ApplyFormatter(msg *core.Message) error {
+	content := format.GetAppliedContent(msg)
+
+	values := make(tcontainer.MarshalMap)
+	err := json.Unmarshal(content, &values)
+
+	if err != nil {
+		format.Logger.Warningf("JSON parser error: %s, Message: %s", err, content)
+		return err
+	}
+
+	var timestamp int64
+	if val, ok := values[format.timeField]; ok {
+		timestamp, err = strconv.ParseInt(val.(string), 10, 64)
+		if err != nil {
+			format.Logger.Error("Invalid time format in message:", err)
+		}
+		delete(values, format.timeField)
+	} else {
+		timestamp = time.Now().Unix()
+	}
+
+	var measurement string
+	if val, ok := values[format.measurement]; ok {
+		measurement = val.(string)
+		delete(values, format.measurement)
+	} else {
+		return fmt.Errorf("Required field for measurement (%s) not found in payload", format.measurement)
```

Comment:
I would turn this around to improve readability / put more emphasis on the early out.
```go
val, ok := values[format.measurement]
if !ok {
    return fmt.Errorf("Required field for measurement (%s) not found in payload", format.measurement)
)
measurement := val.(string)
delete(values, format.measurement)
```